src/pages/EditPlayer/index.js

```diff
@@ -0,0 +1,337 @@
+        avatarToUpload,
+        croppedAreaPixels,
+      )
+
+      dispatch(addPlayerCache({
+        player: {
+          avatarUrl: URL.createObjectURL(croppedImage),
+        }
+      }));
+      setAvatarToUpload(null);
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
+  const readFile = (file) => {
+    return new Promise((resolve) => {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => resolve(reader.result), false);
+      reader.readAsDataURL(file);
+    })
+  };
+
+  const onUploadPlayerAvatar = async (e) => {
+    if (e.target.files && e.target.files.length > 0) {
+      const file = e.target.files[0]
+      let imageDataUrl = await readFile(file)
+      setAvatarToUpload(imageDataUrl)
+    }
+  }
+
+  const onSelectPlayerAvatar = () => {
+    fileInputRef.current.click();
+  };
+
+  const onInputChange = (e) => {
+    const value = e.target.value;
+    const name = e.target.name;
+
+    dispatch(addPlayerCache({
+      player: {
+        [name]: value
+      }
+    }));
+
+    const state = validator(name, value, 'change', playerToEdit, true);
+    setFormState(prev => ({
+      ...prev,
+      [name]: state
+    }))
+  };
+
+  const onInputBlur = (e) => {
+    const value = e.target.value;
+    const name = e.target.name;
+
+    const state = validator(name, value, 'blur', playerToEdit, true);
+    setFormState(prev => ({
+      ...prev,
+      [name]: state
+    }))
+  };
+
+  const onRandomNumber = () => {
+    const randomNumber = Math.floor(Math.random() * 100);
+    dispatch(addPlayerCache({
+      player: {
+        number: randomNumber
+      }
+    }));
+
+    const state = validator('playerNumber', randomNumber, 'blur', playerToEdit, true);
+    setFormState(prev => ({
+      ...prev,
+      playerNumber: state
+    }))
+  };
+
+  const onUpdateNewPlayer = async () => {
+    let valid = true;
+    for (const field of ['name', 'number']) {
+      const fieldState = validator(field, playerToEdit[field], 'blur', playerToEdit, true);
+      setFormState(prev => ({
+        ...prev,
+        [field]: fieldState
+      }))
+
+      if (fieldState.state === 'error') {
+        valid = false;
+      }
+    }
+
+    if (valid) {
+      try {
+        setUpdatePlayerLoading(true);
+
+        const payload = {
+          player: {
+            id: playerToEdit.id,
+            name: playerToEdit.name,
+            number: parseInt(playerToEdit.number),
+            createdBy: playerToEdit.createdBy,
+          }
+        };
+
+        if (croppedImageUrl) {
+          payload.image = croppedImageUrl;
+        }
+
+        console.log('UPLOAD: ', payload)
+
+        const newPlayer = await editPlayerApi(payload);
+        dispatch(editPlayer({ player: newPlayer }));
+        setUpdatePlayerLoading(false);
+        navigate(-1);
+      } catch (err) {
+        console.log('ON CREATE PLAYER API ERR: ', err);
+        setUpdatePlayerLoading(false);
```

Comment:
} catch (err) {
  toast.error('Failed to update player. Please try again.');
  setUpdatePlayerLoading(false);
}